### PR TITLE
TSQL dialect: add parsing for MERGE statement

### LINKED ISSUE #1982: TSQL: PRS error: add possibility to parse MERGE statement
Please add possibility to parse MERGE statement in TSQL dialect.
[merge-transact-sql](https://docs.microsoft.com/ru-ru/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15)

I see MERGE has been implemented already in Exasol and Snowflake dialects of SQLFluff. 
But MERGE  was not included in ANSI base dialect for some reason, although MERGE is in ANSI standard:
[Merge_(SQL)](https://en.wikipedia.org/wiki/Merge_(SQL))

Script example:
```
merge 
	schema1.table1 dst
using
	schema1.table1 src
on
	src.rn = 1
	and dst.e_date_to is null
	and dst.cc_id = src.cc_id
when matched
	then update
		set
			dst.l_id = src.l_id
			, dst.e_date_to = src.e_date_from;
```

### PR DESCRIPTION
**Pull request by tkachenkomaria244**
### Brief summary of the change made
closes #1982 
MergeStatementSegment (and related sub-segments) and OutputClauseSegment were created for MERGE statement.

### Are there any other side effects of this change that we should be aware of?
For special column "$action" used in MERGE <output_clause> I've created "VariableIdentifierSegment" with lexer matcher "var_prefix". I'm not sure this is correct.
[merge-transact-sql](https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15)
[output-clause-transact-sql](https://docs.microsoft.com/en-us/sql/t-sql/queries/output-clause-transact-sql?view=sql-server-ver15)

### Pull Request checklist
- [+] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).

### CODE REVIEW

**Review by tunetheweb — APPROVED** [review]
Quite a lot in there, and don't know TSQL but LGTM over all!

**Comment by tunetheweb** [comment]
@jpers36 any comments before I merge this?

File: src/sqlfluff/dialects/dialect_tsql.py [hunk 1]
```diff
@@ -2767,3 +2778,209 @@ class CreateSchemaStatementSegment(BaseSegment):
...
+                Ref("AliasExpressionSegment", optional=True),
+                Ref("BracketedColumnReferenceListGrammar", optional=True),
+            ),
+        ),
+        Dedent,
+        Ref("JoinOnConditionSegment"),
+        AnyNumberOf(
+            Ref("MergeMatchedClauseSegment"),
+            Ref("MergeNotMatchedClauseSegment"),
+        ),
```
Comment (jpers36) [inline src/sqlfluff/dialects/dialect_tsql.py:2832]: Can we throw a min_times=1 in here?  I believe there needs to be at least one of these clauses.

https://docs.microsoft.com/en-us/sql/t-sql/statements/merge-transact-sql?view=sql-server-ver15#remarks

"At least one of the three MATCHED clauses must be specified, but they can be specified in any order."

Comment (tkachenkomaria244) [reply src/sqlfluff/dialects/dialect_tsql.py:2833]: Added as suggested, thanks

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**